contracts/actions/exchange/LimitSell.sol

```diff
@@ -0,0 +1,155 @@
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public virtual override payable returns (bytes32) {
+        Params memory params = parseInputs(_callData);
+
+        params.exchangeData.srcAddr = _parseParamAddr(
+            params.exchangeData.srcAddr,
+            _paramMapping[0],
+            _subData,
+            _returnValues
+        );
+        params.exchangeData.destAddr = _parseParamAddr(
+            params.exchangeData.destAddr,
+            _paramMapping[1],
+            _subData,
+            _returnValues
+        );
+
+        params.exchangeData.srcAmount = _parseParamUint(
+            params.exchangeData.srcAmount,
+            _paramMapping[2],
+            _subData,
+            _returnValues
+        );
+        params.from = _parseParamAddr(params.from, _paramMapping[3], _subData, _returnValues);
+        params.to = _parseParamAddr(params.to, _paramMapping[4], _subData, _returnValues);
+
+        (uint256 exchangedAmount, bytes memory logData) = _dfsSell(params.exchangeData, params.from, params.to, params.gasUsed);
+        emit ActionEvent("LimitSell", logData);
+        return bytes32(exchangedAmount);
+    }
+
+    /// @inheritdoc ActionBase
+    /// @dev No direct action as it's a part of the limit order strategy
+    function executeActionDirect(bytes memory _callData) public virtual override payable   {}
+
+    /// @inheritdoc ActionBase
+    function actionType() public virtual override pure returns (uint8) {
+        return uint8(ActionType.STANDARD_ACTION);
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+
+    /// @notice Sells a specified srcAmount for the dest token
+    /// @param _exchangeData DFS Exchange data struct
+    /// @param _from Address from which we'll pull the srcTokens
+    /// @param _to Address where we'll send the _to token
+    /// @param _gasUsed Gas used for this strategy so we can take the fee
+    function _dfsSell(
+        ExchangeData memory _exchangeData,
+        address _from,
+        address _to,
+        uint256 _gasUsed
+    ) internal returns (uint256, bytes memory) {
+        // if we set srcAmount to max, take the whole proxy balance
+        if (_exchangeData.srcAmount == type(uint256).max) {
+            _exchangeData.srcAmount = _exchangeData.srcAddr.getBalance(address(this));
+        }
+
+        // Validate price that is set in the trigger
+        uint256 currPrice = uint256(tempStorage.getBytes32("CURR_PRICE"));
+        require(currPrice > 0, "LimitSell: Price not set");
+
+        // Reset the current price for the next strategy
+        tempStorage.setBytes32("CURR_PRICE", bytes32(0));
+        _exchangeData.dfsFeeDivider = 0;
+
+        if (_exchangeData.minPrice != currPrice) {
+            revert WrongPriceFromTrigger(currPrice, _exchangeData.minPrice);
+        }
+     
+        _exchangeData.srcAddr.pullTokensIfNeeded(_from, _exchangeData.srcAmount);
```

Comment:
if we write 
_exchangeData.srcAmount = _exchangeData.srcAddr.pullTokensIfNeeded(_from, _exchangeData.srcAmount);
we can delete lines 86-88

Reply:
Good point, as the change is not critical and already deployed, will leave it as is for now.